src/routes/servers/server_edit.rs

```diff
@@ -1,159 +1,162 @@
-use crate::notifications::events::ClientboundNotification;
-use crate::util::result::{Error, Result, EmptyResponse};
-use crate::{database::*, notifications::events::RemoveServerField};
+use revolt_quark::{
+    models::{
+        server::{Category, FieldsServer, PartialServer, SystemMessageChannels},
+        File, Server, User,
+    },
+    perms, Db, Error, Permission, Ref, Result,
+};
 
-use mongodb::bson::{doc, to_bson, to_document};
 use rocket::serde::json::Json;
 use serde::{Deserialize, Serialize};
 use validator::Validate;
 
-#[derive(Validate, Serialize, Deserialize)]
-pub struct Data {
+/// # Server Data
+#[derive(Validate, Serialize, Deserialize, JsonSchema)]
+pub struct DataEditServer {
+    /// Server name
     #[validate(length(min = 1, max = 32))]
     name: Option<String>,
+    /// Server description
     #[validate(length(min = 0, max = 1024))]
     description: Option<String>,
+
+    /// Attachment Id for icon
     icon: Option<String>,
+    /// Attachment Id for banner
     banner: Option<String>,
+
+    /// Category structure for server
+    #[validate]
     categories: Option<Vec<Category>>,
+    /// System message configuration
     system_messages: Option<SystemMessageChannels>,
-    remove: Option<RemoveServerField>,
-    nsfw: Option<bool>,
+
+    // Whether this server is age-restricted
+    // nsfw: Option<bool>,
```

Comment:
```suggestion
    nsfw: Option<bool>,
```

Reply:
I'm not sure if this should be restricted to instance admins, though it could be a non-reversible switch.